src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -2683,21 +2679,32 @@ public boolean canConvert(Class aClass) {
         @Override
         protected void doMarshal(
                 Object source, HierarchicalStreamWriter writer, MarshallingContext context) {
-            GrowableInternationalString internationalString = (GrowableInternationalString) source;
-            Set<Locale> locales = internationalString.getLocales();
-            int size = locales.size();
-            for (Locale l : locales) {
-                if (l != null) {
-                    String elementName = l.toLanguageTag();
-                    if (elementName.contains(" ")) elementName = elementName.replaceAll(" ", "__");
-                    writer.startNode(elementName);
-                    writer.setValue(internationalString.toString(l));
-                    writer.endNode();
-                } else if (size > 1) {
-                    writer.startNode(DEFAULT_LOCALE);
-                    writer.setValue(internationalString.toString(l));
-                    writer.endNode();
+            boolean isGrowable = source instanceof GrowableInternationalString;
+            String defaultValue = "";
+            if (isGrowable) {
+                GrowableInternationalString internationalString =
+                        (GrowableInternationalString) source;
+                Set<Locale> locales = internationalString.getLocales();
+                int size = locales.size();
+                for (Locale l : locales) {
+                    if (l != null) {
+                        String elementName = l.toLanguageTag();
+                        if (elementName.contains(" "))
+                            elementName = elementName.replaceAll(" ", "__");
+                        writer.startNode(elementName);
+                        writer.setValue(internationalString.toString(l));
+                        writer.endNode();
+                    } else if (size > 1) {
+                        writer.startNode(DEFAULT_LOCALE);
+                        writer.setValue(internationalString.toString(l));
+                        writer.endNode();
+                    }
                 }
+            } else {
+                defaultValue = source.toString();
+                writer.startNode(DEFAULT_LOCALE);
+                writer.setValue(defaultValue);
+                writer.endNode();
```

Comment:
I'm a bit confused here: in the closed pr you were using the Locale.getDefault() to start the node. But now you swithced to the DEFAULT_LOCALE constant. Why this change? Note that the DEFAULT_LOCALE would result in an xml element name <default>. It seems to me that the Locale.getDefault() is the preferred option here.